File: JAVA/Root to Leaf Paths.java
class Solution {
    public static ArrayList<ArrayList<Integer>> Paths(Node root) {
    	ArrayList<ArrayList<Integer>> arr=new ArrayList<>();
    	ArrayList<Integer> temp=new ArrayList<>();
    	findPath(root,temp,arr);
    	return arr;
    }
	private static void findPath(Node node, ArrayList<Integer> temp, ArrayList<ArrayList<Integer>> arr) {
		if(node==null)return;
		temp.add(node.data);
		if(node.left==null && node.right==null) arr.add(new ArrayList<>(temp));
		findPath(node.left, temp, arr);
		findPath(node.right, temp, arr);
		temp.remove(temp.size()-1);
	}
}

File: C++/Root to Leaf Paths.cpp
class Solution {
  private:
    void travel(Node* root, vector<vector<int>>&ans, vector<int> curr){
        if(root == NULL){
            return;
        }
        curr.push_back(root->data);
        if(root->left == NULL && root->right == NULL){
            ans.push_back(curr);
            return;
        }
        travel(root->left, ans, curr);
        travel(root->right, ans, curr);
    }
  public:
    vector<vector<int>> Paths(Node* root) {
        vector<vector<int>> ans;
        vector<int> curr;
        travel(root, ans, curr);
        return ans;
    }
};
